float64[2] 	u
float64[4] 	sigma
float64	    z
---
float64[2] 	u
float64[4] 	sigma
